packages/big-design/src/components/StatefulTable/spec.tsx

```diff
@@ -550,6 +551,88 @@ describe('test search in the StatefulTable', () => {
 
     expect(rows).toHaveLength(105);
   });
+
+  test('maintains filtered state when both filters and search are active & items per page is changed', async () => {
+    const filters: StatefulTablePillTabFilter<TestItem> = {
+      pillTabs: [{ title: 'Low stock', id: 'low_stock' }],
+      filter: (_itemId, items) => items.filter((item) => item.stock !== 0 && item.stock < 10),
+    };
+    const { findByText, findAllByRole, getByPlaceholderText, getByText } = render(
+      getSimpleTable({ filters, pagination: true, search: true }),
+    );
+
+    // Click on filter pill
+    fireEvent.click(getByText('Low stock'));
```

Comment:
Can we use `screen.findBy*` and `userEvent.click(...)` utilities instead of using the return methods of `render`? We want to eventually migrate all this code to using RTLs best practices and this will help us.